### PR TITLE
Final submission PR

### PR DESCRIPTION
**Pull request by asharanjith**
The project is a single-page application (SPA) built with React and Redux and using the data from [CoinGecko API](https://www.coingecko.com/en/api/documentation)

This PR is raised to 
- [x]  When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for example by the category name.
- [x]  Along with the category name, you will display some numeric values per category that come from the API.
- [x] When the user clicks (or taps) on a category item, the application navigates to the details page.
- [x] In the details page, the web app retrieves data from the API to show detailed data in the category.
- [x] When the user clicks on the "Back" button, the user navigates to the home page.
- [x] Create integration tests for your application using the React Testing Library.
- [x] Followed gitflow and add linters

[Video presentation](https://drive.google.com/file/d/1KPg_C-jZvKAmM5-MnJ_GkjHMX4YWVwjs/view?usp=share_link)
[Live Demo](https://crypto-currency-9i1l.vercel.app/)

### CODE REVIEW

File: src/components/homePage/homePage.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        {allData
+          .filter((product) => product.name.toLowerCase().includes(search.toLowerCase()))
+          .map((product, index) => (
+            <div
+              className="homePage__product"
+              key={product.id}
+              style={index % 2 === 0 ? { backgroundColor: '#0096c7' } : { backgroundColor: '#00b4d8' }}
+            >
+              {innerwidth > 500 ? (
+                <img
```
Comment (Munsa1) [inline src/components/homePage/homePage.js]: - [ ] Great work on the app, however it is a core requirement that your design matches the given design, this includes the layout meaning `composition and space between elements`. and typography `meaning the font weight of some elements, font face, font size e.t.c`. Kindly implement this

image

Your design
image

Given design
image

File: src/_test_/Navbar.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+describe('Navbar', () => {
+  test('should render correctly', () => {
+    const tree = renderer.create(
+      <BrowserRouter>
+        <NavBar />
+      </BrowserRouter>,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (Munsa1) [inline src/_test_/Navbar.test.js]: - [ ] Kindly add some  unit tests for several components of the app and some pure functions in Redux code, as this is part of the core requirements

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @asharanjith, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by MaganaAsiati — APPROVED** [review]
# Status: Approved [✔️](https://emojipedia.org/check-mark/)
## Hi @asharanjith   , I'm @MaganaAsiati , your reviewer for this PR  :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


clap

## Highlights
- descriptive README file  :heavy_check_mark:👨🏻‍💻
- Use of GitHub flow :heavy_check_mark:👨🏻‍💻
-No linter error  :heavy_check_mark:👨🏻‍💻

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future.
- [ ] [OPTIONAL] kindly  Add a meaningful description in your pull request in the future which gives more information about what you implement in this milestone :heavy_check_mark:👨🏻‍💻


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Cheers and Happy coding!👏👏👏

**Pull request merged**